android/runtime/common/src/java/org/appcelerator/kroll/util/TiTempFileHelper.java

```diff
@@ -167,7 +170,13 @@ private void updateTempDir()
 	{
 		String extState = Environment.getExternalStorageState();
 		if (!extState.equals(previousExternalStorageState)) {
-			if (Environment.MEDIA_MOUNTED.equals(extState)) {
+			if (Environment.MEDIA_MOUNTED.equals(extState)
+				&& (Build.VERSION.SDK_INT < 23
+					|| (Build.VERSION.SDK_INT >= 23
```

Comment:
Maybe we can get away with removing `Build.VERSION.SDK_INT >= 23` check ? It should be evaluated only if the first check `Build.VERSION.SDK_INT < 23` is `false`, which will leave us with it always being `true`.

Reply:
What if `Environment.MEDIA_MOUNTED.equals(extState)` is `false`?

Reply:
If `Environment.MEDIA_MOUNTED.equals(extState) === false`, then yes the rest of the `if` won't get evaluated. I believe Yordan is correct and that you could remove `Build.VERSION.SDK_INT >= 23` since it is implied (being the exact opposite of `Build.VERSION.SDK_INT < 23`)